### DISCUSSION TITLE
Minify flag on Tailwind CLI creating invalid CSS for Safari 14

### DISCUSSION THREAD

**Discussion by DeveloperRob** [tailwindlabs/tailwindcss#12140] [Help]
Good Afternoon all,

We had a bit of a tricky to diagnosis issue on Safari 14 on iOS recently. We are currently creating a minified CSS file at present using `npx tailwindcss --minify`. Inspecting the non-minified CSS, near the top of the file you have as expected (most of the variables have been replaced for brevity, but the removed variables are the same in both set of rules):
```
*, ::before, ::after {
  --tw-border-spacing-x: 0;
  --tw-border-spacing-y: 0;
...
  --tw-backdrop-saturate:  ;
  --tw-backdrop-sepia:  ;
}

::backdrop {
  --tw-border-spacing-x: 0;
  --tw-border-spacing-y: 0;
..
  --tw-backdrop-saturate:  ;
  --tw-backdrop-sepia:  ;
}
```

The problem I have is when using the minified CSS, the rules are combined - so you have:
```
*,::backdrop,:after,:before{--tw-border-spacing-x:0;--tw-border-spacing-y:0;...
```

`::backdrop` was only introduced in Safari 15.4 (https://caniuse.com/dialog), so when the rules are combined Safari considers the rules invalid and ignores them. This then means anything that relies on these default variables no longer works correctly.

I believe --minify uses `cssnano`, but when testing with that allow it doesn't combine the rules. Is this a bug in Tailwind, or am I missing a step in the process which is responsible for incorrect combining the rules?

See https://github.com/tailwindlabs/tailwindcss/discussions/8788 where the problem was first found; with the issue in the non-minifed output being fixed in https://github.com/tailwindlabs/tailwindcss/pull/8567.

Example to replicate the issue:
`index.html:`
```
<!doctype html>
<html>
	<head>
		<title>Test</title>
		<link rel="stylesheet" href="output.css">
		<style>
			.plainCss{
				height: 80px;
				width: 80px;
				background-image: linear-gradient(to right, #991b1b, #166534);
				border-width: 2px;
				border-color: black;
			}
		</style>
	</head>
	<body>
		<div class="border-2 border-black h-20 w-20 bg-gradient-to-r from-red-800 to-green-800">
		</div>
		<br><br>
		<div class="border-2 border-black h-20 w-20 bg-gradient-to-r from-red-800 from-0% to-green-800 to-100%">
		</div>
		<br><br>
		<div class="plainCss">
		</div>
	</body>
</html>
```
`tailwind.config.js`:
```
module.exports = {
  content: [
    'index.html'
  ]
}
```

**Working:**
`npx tailwindcss -o output.css`
**Doesn't work (the first gradient box is empty):**
`npx tailwindcss -o output.css --minify`

Thanks,
Robert

**Comment by e2corporation** [comment]
I've observed the same behavior with non-functional gradients with minified build using Tailwinds CLI and **Safari v14.1.1** on **Mac OS Big Sur**

**Comment by KristofersOzolinsMagebit** [comment]
Just experienced this issue. The fix was to install and add `cssnano` to postcss.config.js
